### PR TITLE
Issue-5: Add initial support for List extras

### LINKED ISSUE #5: Add initial support for List extras
Must include the following:

* The following static methods:
    * `append :: [a] → [a] → [a]`: Appends two lists.
    * `foldLeft :: Foldable t => (b → a → b) → b → t a → b`: Left-associative fold of a structure.
    * `foldRight :: Foldable t => (a → b → b) → b → t a → b`: Right-associated fold of a structure.
    * `head :: [a] → a`: Extracts the first element of a list which must be non-empty.
    * `init :: [a] → [a]`: Extracts all of the elements before the last of a list which must be non-empty.
    * `isEmpty :: Foldable a → Boolean`: Determines whether or not the list (read `Foldable`) is empty.
    * `length :: Foldable a → Number`: Returns the size/length of a finite structure.
    * `last :: [a] → a`: Extracts the last element of a list which must be non-empty.
    * `map :: (a → b) → [a] → [b]`: Maps the underlying values within a list.
    * `tail :: [a] → [a]`: Extracts the elements after the head of a list which must be non-empty.
    * `uncons :: [a] → Maybe (a, [a])`: Decomposes a list into its head and tail. Returns `Nothing` if the list is empty; otherwise a `Just (x, xs)` where `x` is the head of the list and `xs` is the tail.

### PR DESCRIPTION
**Pull request by jlmorgan**
closes #5
SemVer:Minor
enhancement

### CODE REVIEW

File: docs/data/Validation/static/validate.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # `Validation.<F, S>validate(Predicate<S> predicate, F failureValue, S value)`
 
-Validates a value `b` and returns a `Success` of `b` if the `predicate` returns `true`; otherwise, a `Failure` of `a`.
+Validates a value `b` and A `Success` of `b` if the `predicate` returns `true`; otherwise, a `Failure` of `a`.
```
Comment (JeremiahSanders) [inline docs/data/Validation/static/validate.md]: Double-check casing.

File: src/main/java/com/github/jlmorgan/data/Validation.java [hunk 2]
```diff
@@ -218,7 +218,7 @@
   }
 
   /**
-   * Validates a value {@code b} and returns a {@code Success} of {@code b} if the {@code predicate} returns
+   * Validates a value {@code b} and A {@code Success} of {@code b} if the {@code predicate} returns
```
Comment (JeremiahSanders) [inline src/main/java/com/github/jlmorgan/data/Validation.java]: Double-check casing of change.

File: src/main/java/com/github/jlmorgan/data/Maybe.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
  */
 public interface Maybe<A> {
   /**
-   * Takes a list of {@link Maybe} and returns a list of the {@code Just} values.
+   * Takes a list of {@link Maybe} and A list of the {@code Just} values.
```
Comment (JeremiahSanders) [inline src/main/java/com/github/jlmorgan/data/Maybe.java]: Double-check casing of new text.

**Review by JeremiahSanders — APPROVED** [review]

File: docs/data/Validation/static/validate.md (hunk 1 above)
Comment (jlmorgan) [reply docs/data/Validation/static/validate.md]: Good catches. Pushed a `fixup` with the changes for the casing.

**Review by doomchild — APPROVED** [review]
Seems legit.

**Review by JeremiahSanders — APPROVED** [review]

**Pull request merged**